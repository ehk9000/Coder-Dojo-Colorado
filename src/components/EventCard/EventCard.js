import React from 'react';
import PropTypes from 'prop-types';

function EventCard({ event }) {
  const { 
    name, notes, startTime, endTime, id 
  } = event;
  const startDate = new Date(startTime);
  const endDate = new Date(endTime);
  const timeSetting = { hour: 'numeric', hour12: true };

  return (
    <article>
      <h3>{name}</h3>
      <p>
        When:
        {startDate.toDateString()}
      </p>
      <p>
        {startDate.toLocaleString('en-US', timeSetting)} 
        - 
        {endDate.toLocaleString('en-US', timeSetting)}
      </p>
      <p>{notes}</p>
    </article>
  );
}

export default EventCard;

EventCard.propTypes = {
  event: PropTypes.object
};

EventCard.defaultProps = {
  event: {}
};
