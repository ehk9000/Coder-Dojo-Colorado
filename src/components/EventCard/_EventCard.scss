.EventCard {
  padding: 5% 8% 10% 8%;
  border: 2px solid #faa31a;
  margin: 10px;
  display: flex;
  flex-direction: column;
  align-items: space-between;
  position: relative;

  .details {
    display: flex;

    h3 {
      color: #298244;
      flex: 2;
    }
    
    .date-container {
      flex: 1;

      .hours {
        font-size: .8rem;
      }
    }
  }

  .notes {
    position: absolute;
    bottom: 0;
    left: 0;
    padding: 10px 0;
    text-indent: 20px;
    font-size: .8rem;
    width: 100%;
    background-color: #f4f5f6;
  }

}