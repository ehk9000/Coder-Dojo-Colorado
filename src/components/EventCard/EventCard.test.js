import React from 'react';
import { shallow } from 'enzyme';
import EventCard from './EventCard';

describe('EventCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<EventCard />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
