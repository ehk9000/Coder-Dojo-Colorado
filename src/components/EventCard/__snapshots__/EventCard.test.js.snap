// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard should match snapshot 1`] = `
<article>
  <h3 />
  <p>
    When:
    Invalid Date
  </p>
  <p>
    Invalid Date
    -
    Invalid Date
  </p>
  <p />
</article>
`;
